pallets/messages/src/lib.rs

```diff
@@ -138,24 +146,23 @@ pub mod pallet {
 
 			// TODO: validate schema existence and validity from schema pallet
 
-			let mut block_messages = <BlockMessages<T>>::get();
+			let mut block_messages = <BlockMessages<T>>::get().into_inner();
 			let current_size: u16 = block_messages
 				.len()
 				.try_into()
 				.map_err(|_| Error::<T>::TypeConversionOverflow)?;
-			ensure!(
-				current_size < T::MaxMessagesPerBlock::get(),
-				Error::<T>::TooManyMessagesInBlock
-			);
 
 			let m = Message {
-				data: message,
+				data: message.try_into().map_err(|_| Error::<T>::TooLargeMessage)?,
```

Comment:
I think you already have this validation above? Why do you need it again in here?

Reply:
Redundant. Removed!